src/components/tag/index.tsx

```diff
@@ -2,10 +2,14 @@ import { IconX16 } from '@hashicorp/flight-icons/svg-react/x-16'
 import { TagProps } from './types'
 import s from './tag.module.css'
 
-const Tag = ({ text, onRemove }: TagProps) => {
+const Tag = ({ text, onRemove, buttonAriaLabel }: TagProps) => {
 	return (
 		<span className={s.root}>
-			<button className={s.removeButton} onClick={onRemove}>
+			<button
+				className={s.removeButton}
+				onClick={onRemove}
+				aria-label={buttonAriaLabel ?? `Dismiss ${text}`}
```

Comment:
based on the HDS implementation, the `aria-label` appears to default to `Dismiss ${text}`. I also added the `buttonAriaLabel` prop to provide additional flexibility here.